components/summary-detail/index.jsx

```diff
@@ -0,0 +1,165 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Welcome Mat design pattern](https://lightningdesignsystem.com/components/welcome-mat/) in React.
+// Based on SLDS v2.4.0
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+
+import assign from 'lodash.assign';
+
+import Icon from '../icon';
+
+import { SUMMARY_DETAIL } from '../../utilities/constants';
+
+const displayName = SUMMARY_DETAIL;
+
+const propTypes = {
+	assistiveText: PropTypes.shape({
+		toggleButton: PropTypes.string,
+	}),
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Title for the summary
+	 */
+	title: PropTypes.string,
+	/**
+	 * Summary in detail
+	 */
+	detail: PropTypes.string,
+	/**
+	 * Sub-title for the summary.
+	 */
+	subTitle: PropTypes.string,
+	/**
+	 * Text to be shown on badge on right of the title
+	 */
+	badgeText: PropTypes.string,
```

Comment:
This should take a whole badge component. Since we don't have one yet, please make it badge and pass HTML for now. that way folks can control the theme, etc.